cli/tests/integration/mod.rs

```diff
@@ -1202,26 +1206,29 @@ async fn listen_tls_alpn_fail() {
       let msg = std::str::from_utf8(&buffer).unwrap();
       assert_eq!(msg, "READY");
 
-      let mut cfg = rustls::ClientConfig::new();
-      let reader = &mut BufReader::new(Cursor::new(include_bytes!(
+      let mut reader = &mut BufReader::new(Cursor::new(include_bytes!(
         "../testdata/tls/RootCA.crt"
       )));
-      cfg.root_store.add_pem_file(reader).unwrap();
+      let certs = rustls_pemfile::certs(&mut reader).unwrap();
+      let mut root_store = rustls::RootCertStore::empty();
+      root_store.add_parsable_certificates(&certs);
+      let mut cfg = rustls::ClientConfig::builder()
+        .with_safe_defaults()
+        .with_root_certificates(root_store)
+        .with_no_client_auth();
       cfg.alpn_protocols.push("boofar".as_bytes().to_vec());
       let cfg = Arc::new(cfg);
 
-      let hostname =
-        webpki::DNSNameRef::try_from_ascii_str("localhost").unwrap();
-
+      let hostname = rustls::ServerName::try_from("localhost").unwrap();
       let tcp_stream = tokio::net::TcpStream::connect("localhost:4505")
         .await
         .unwrap();
       let mut tls_stream =
-        TlsStream::new_client_side(tcp_stream, &cfg, hostname);
+        TlsStream::new_client_side(tcp_stream, cfg, hostname);
       tls_stream.handshake().await.unwrap();
```

Comment:
This call fails with:
```
thread 'integration::listen_tls_alpn_fail' panicked at 'called `Result::unwrap()` on an `Err` value: Custom { kind: UnexpectedEof, error: "tls handshake eof" }', cli/tests/integration/mod.rs:1228:36
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

@piscisaureus I'm a bit confused here, why did handshake previously succeeded if there were no matching ALPN protocols?

Reply:
> @piscisaureus I'm a bit confused here, why did handshake previously succeeded if there were no matching ALPN protocols?

I don't know, I also think `handhake()` should fail if there are no matching ALPN protocols.
I've updated the tests to reflect that.